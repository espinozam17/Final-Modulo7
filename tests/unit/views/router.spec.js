// tests/unit/views/router.spec.js
import { shallowMount, mount } from "@vue/test-utils";
import { createRouter, createWebHistory } from "vue-router";
import HomeView from "@/views/HomeView.vue";
import AboutView from "@/views/AboutView.vue";

describe("Componente HomeView.vue", () => {
  test("Validaci贸n de match con el snapshot", () => {
    //Selecci贸n el Sujeto de pruebas
    const wrapper = shallowMount(HomeView);

    //Aserci贸n
    expect(wrapper.html()).toMatchSnapshot();
  });
});
it("Test - Probar la existencia del componente en la ruta", async () => {
  const routes = [
    {
      path: "/",
      name: "home",
      component: () => HomeView,
    },
  ];
  const router = createRouter({
    history: createWebHistory(process.env.BASE_URL),
    routes,
  });

  router.push("/");
  await router.isReady();

  const wrapper = mount(HomeView, {
    global: {
      plugins: [router],
    },
  });

  expect(wrapper.findComponent(HomeView).exists()).toBe(true);
});

describe("AboutView.vue", () => {
  it("renders About page", () => {
    const wrapper = shallowMount(AboutView);
    expect(wrapper.text()).toMatch("This is an about page");
    //Aserci贸n
    expect(wrapper.html()).toMatchSnapshot();
  });
});
it("Test - Probar la existencia del componente en la ruta", async () => {
  const routes = [
    {
      path: "/about",
      name: "about",
      component: () => AboutView,
    },
  ];
  const router = createRouter({
    history: createWebHistory(process.env.BASE_URL),
    routes,
  });

  router.push("/");
  await router.isReady();

  const wrapper = mount(AboutView, {
    global: {
      plugins: [router],
    },
  });

  expect(wrapper.findComponent(AboutView).exists()).toBe(true);
});
