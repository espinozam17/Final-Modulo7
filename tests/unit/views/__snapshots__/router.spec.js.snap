// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView.vue renders About page 1`] = `
<div class="about">
  <h1>This is an about page</h1>
</div>
`;

exports[`Componente HomeView.vue Validaci√≥n de match con el snapshot 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <hello-world-stub msg="Welcome to Your Vue.js App"></hello-world-stub>
</div>
`;
